connect_four.py

```diff
@@ -25,48 +29,53 @@ def print_board(self) -> None:
 
             print(print_row)
             row_count -= 1
-        
-        print("  A  B  C  D  E  F  G")  #TODO needs to flex to board size
+
+        print_row = " "
+        for lett in range(self.num_cols):
+            print_row += " " + chr(lett+65) + " "
+        print(print_row)
 
 
     #function for checking the validity of a chosen column. Checks if col exists and is not full. 
-    def check_col(self, col: str): 
-        while True:        
-            if col not in self.col_dict: 
-                print(player1 + ", that column does not exist. Please pick another column (A-G):")
+    def check_col(self, col: str) -> int: 
+        while True:   
```

Comment:
Avoid having the true loop in the check col, this can be a part of the game loop. You don't want the board to have to handle user input.